src/status_im/ui/screens/browser/events.cljs

```diff
@@ -58,6 +60,21 @@
      (-> (add-browser-fx cofx new-browser)
          (update-in [:db :browser/options] #(assoc % :browser-id (:browser-id new-browser)))))))
 
+(handlers/register-handler-fx
+ :update-browser-with-history
+ [re-frame/trim-v]
+ (fn [{:keys [now] :as cofx} [browser url]]
+   (let [history-index (if (nil? (:history-index browser))
+                         0
+                         (:history-index browser))
+         history (:history browser)
+         history-url (nth history history-index)
+         browser-history-to-index (drop history-index history)
+         new-history (if (not= history-url url) (conj browser-history-to-index url) history)
+         new-browser (get-new-browser (assoc browser :history new-history :history-index 0) now)]
+     (-> (add-browser-fx cofx new-browser)
+         (update-in [:db :browser/options] #(assoc % :browser-id (:browser-id new-browser)))))))
```

Comment:
`(update-in [:db :browser/options] assoc :browser-id (:browser-id new-browser)))))))`

Reply:
Done!